### DISCUSSION TITLE
Deleting big amount of data causing connection error

### DISCUSSION THREAD

**Discussion by bmajkut** [prisma/prisma#13480] [Q&A] [answered]
Hello!
I am working on a project using Nest.js and Prisma orm. One of the feature in that project is about deleting old data from two tables. When I am trying to use deleteMany with a basic filter I am getting connection error.
I assume its because of really big amount of data to delete but how should I perform such operation? I turned on query logging and I saw that simple delete function is firing two queries:
first: fetching ids of rows to delete
second: deleting those particular rows
In case we want to delete i.e. 100k rows the query would be extremely long.
Should I use a raw SQL query for such operation or am I doing something wrong?

**Comment by nurul3101** [comment, marked as answer]
Hey @bmajkut 👋 
Are you trying this on the latest `@prisma/client` version?
Are any of the records getting deleted or you straight away get the connection error?

**Reply by bmajkut** [reply]
Nothing is being deleted. Right now I've tried with "$executeRaw" and I've observed that when I want to deleted i.e. 40k rows with query like this:
`DELETE FROM public."StateAtTimePoint" WHERE "coordinateId" in (${Prisma.join(coordsIdsToRemove)})`
I get an error and it is related to length of `coordsIdsToRemove`. When there are too many id in that variable database is throwing "N/A" error. I have to splice that array to smaller chunks and run multiple queries.

**Reply by nurul3101** [reply]
Could you please create a [Bug Report](https://github.com/prisma/prisma/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.yml) for this, so that we could have a look and fix it?

**Reply by bmajkut** [reply]
Will do, but first I will prepare minimum code

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
